### PR TITLE
fix: Always use frontendGeneratedFolder for generated files

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

Currently some node tasks hard-code the generated folder as child
of the frontend directory, whereas other tasks uses the
frontendGeneratedFolder configuration.
This change makes all tasks use the value of frontendGeneratedFolder.

Part of #12880

## Type of change

- [X] Bugfix
- [ ] Feature

## Checklist

- [X] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [X] I have added a description following the guideline.
- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt [hunk 1]
```diff
@@ -143,7 +143,7 @@ public open class VaadinFlowPluginExtension(project: Project) {
     /**
      * The folder where flow will put TS API files for client projects.
      */
-    public var generatedTsFolder: File = File(project.projectDir, "frontend/generated")
+    public var generatedTsFolder: File? = null
```
Comment (taefi) [inline flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt:146]: Why defer the initialization? It is only dependent on the `frontendDirectory` which has already been initialized and has not changed. 
```suggestion
    public var generatedTsFolder: File = File(frontendDirectory, FrontendUtils.GENERATED)
```

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskGenerateBootstrapTest.java [hunk 2]
```diff
@@ -98,6 +102,20 @@ public void should_importFrontendIndexTS()
         Assert.assertTrue(content.contains("import '../index';"));
     }
 
+    @Test
+    public void should_importFrontendIndexTS_customFrontendFolder()
```
Comment (taefi) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskGenerateBootstrapTest.java:106]:
```suggestion
    public void should_importFrontendIndexTS_fromCustomFrontendFolder()
```

File: flow-tests/test-custom-frontend-directory/test-themes-custom-frontend-directory/pom-generatedTsDir.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        <phase>initialize</phase>
+                        <configuration>
+                            <directory>${project.basedir}/frontend</directory>
+                            <failOnError>false</failOnError>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build></project>
```
Comment (taefi) [inline flow-tests/test-custom-frontend-directory/test-themes-custom-frontend-directory/pom-generatedTsDir.xml:83]:
```suggestion
    </build>
</project>

```

**Review by taefi — CHANGES_REQUESTED** [review]

**Review by taefi — APPROVED** [review]

**Comment by mcollovati** [comment]
Closing as superseded by #14144

**Pull request closed without merging**